### ISSUE TITLE
setItemIndex method not updating the current positions

### ISSUE THREAD

**Issue by abhisb** [openseadragon/openseadragon#1130]
`setItemIndex` method of World class is not updating positions of the items. Index of items are setting properly, but positions aren't changing. Positions(**x and y coordinates**) still remain the same on checking `viewer.world.getItemAt(i).getBounds()`. I tried to set positions of the items manually before splice the array and it's positions are changing. I would like to know if there is any way to set positions dynamically along with changing item indexes.

 In the code it's just using splice method to change the index of the items and positions are not being affected.
```

this._items.splice( oldIndex, 1 );
this._items.splice( index, 0, item );
```

**Comment by iangilman** [comment]
I assume you're using collection mode and you expect the collection to rearrange itself when setItemIndex is called? That's an excellent point. Looks like we also don't arrange after removing an item from the world either. Both of these should be fixed (but just for collection mode, of course). 

Would you be up for making a patch to fix this? Probably the thing to do would be to have the Viewer watch for `remove-item` and `item-index-change` events on its World and then call the World's `arrange` function with all of the viewer's settings similar to here:

https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L1310

Meanwhile, a workaround is to call `arrange` directly after you call `setItemIndex`.

**Comment by abhisb** [comment]
Yeah, I'm using collection mode. I have tried the above ways but nothing did work. `arrange` function after `setItemIndex` simply setting position `x`  value of all items to 0, so all items are getting placed vertical at the center of the viewport.

Is there are a function or way to reset positions of world data (`this._items`) again?

**Comment by iangilman** [comment]
Sounds like you're calling `arrange` without the collection settings you've given to the viewer. If you pass those settings into `arrange`, like is done here:

https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L1310

...it should give you the correct results.
